File: linechart/src/main/java/com/davidmiguel/linechart/formatter/PercentYAxisValueFormatter.java
package com.davidmiguel.linechart.formatter;

import android.graphics.RectF;

import androidx.annotation.NonNull;

import java.text.DecimalFormat;
import java.text.NumberFormat;

/**
 * Formatter that adds a % sign after the number.
 */
public class PercentYAxisValueFormatter implements YAxisValueFormatter {

    private NumberFormat formatter;

    public PercentYAxisValueFormatter() {
        formatter =  new DecimalFormat( "#.##'%'");
    }

    @NonNull
    @Override
    public String getFormattedValue(float value, @NonNull RectF dataBounds, int gridYDivisions) {
        return formatter.format(value);
    }
}

File: linechart/src/main/java/com/davidmiguel/linechart/formatter/PercentYAxisValueFormatter.kt
package com.davidmiguel.linechart.formatter

import android.graphics.RectF
import java.text.DecimalFormat
import java.text.NumberFormat

/**
 * Formatter that adds a % sign after the number.
 */
class PercentYAxisValueFormatter : YAxisValueFormatter {

    private val formatter: NumberFormat = DecimalFormat("#.##'%'")

    override fun getFormattedValue(value: Float, dataBounds: RectF, gridYDivisions: Int): String {
        return formatter.format(value.toDouble())
    }
}
